test/e2e/common/node/pod_admission.go

```diff
@@ -65,6 +66,49 @@ var _ = SIGDescribe("PodOSRejection", framework.WithNodeConformance(), func() {
 	})
 })
 
+var _ = SIGDescribe("PodOSRejectionStatus", func() {
+	f := framework.NewDefaultFramework("pod-os-rejection-status")
+	f.NamespacePodSecurityLevel = admissionapi.LevelBaseline
+	ginkgo.Context("Kubelet", func() {
+		ginkgo.It("should reject pod when the node OS doesn't match pod's OS", func(ctx context.Context) {
+			linuxNode, err := findLinuxNode(ctx, f)
+			framework.ExpectNoError(err)
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wrong-pod-os",
+					Namespace: f.Namespace.Name,
+				},
+				Spec: v1.PodSpec{
+					OS: &v1.PodOS{
+						Name: "windows", // explicitly set the pod OS to a wrong but valid value
```

Comment:
if we will make this test to be a conformance test, it will need to run on Windows =). I like the idea though. Maybe detect whether the test is running with Windows nodes by checking the node status

Reply:
> Alternatively mark the test with the "LinuxOnly"

Should This label be applied to https://github.com/carlory/kubernetes/blob/fix-126662-kubelet/test/e2e/common/node/pod_admission.go#L36 too?

Reply:
yeah, seems like it

Reply:
I found the test has a check that if no linux node is found, the test will be skipped. https://github.com/carlory/kubernetes/blob/fix-126662-kubelet/test/e2e/common/node/pod_admission.go#L131 Do we still need to add this `LinuxOnly` feature label?

Reply:
updated

Reply:
@SergeyKanzhelev I changed the new test. It can run on any worker node. so I drop the `LinuxOnly ` from the new test, but still added it to the existing test.